### PR TITLE
Bugs/nrhp preference address geocoding #163639900

### PR DESCRIPTION
**Pull request by callaghanc**
https://www.pivotaltracker.com/story/show/163639900

Remaining tasks on this PR:
- [x] Decide whether to use existing "no candidates found" email error notification when candidates are found but none are acceptable, or to create new one. - **Decision:** use existing notification
- [x] Add tests.
- [x] Decide what our policy is about what addresses we can include in the test - i.e. are there any privacy or other types of concerns around the inclusion of these addresses in our tests.
- [ ] Test the new candidate selection process against the list of addresses provided by Charlie in the comments of the Pivotal story.

### CODE REVIEW

**Comment by callaghanc** [comment]
@akegan This is ready for review.

File: Gemfile.lock [hunk 1]
```diff
@@ -438,7 +440,7 @@ DEPENDENCIES
   pg
   phantomjs (~> 2.1.1)
   prerender_rails (~> 1.6.0)
-  pry
+  pry-byebug
```
[the remark this answers is not in the document]
Comment (callaghanc) [reply Gemfile.lock]: pry provides an enhanced IRB shell and provides the ability to pause execution and open one of those advanced shells to have a poke around at the pause point. it doesn't allow navigation through the code execution. byebug is a debugging gem providing standard stuff like breakpoints and being able to step into and over instructions and other debugging commands. pry-byebug combines the two to allow the use of the debugging commands with the enhanced pry shell.

[the remark this answers is not in the document]
Comment (akegan) [reply Gemfile.lock]: cool! I'd love to learn how to use it at some point!

**Comment by callaghanc** [comment]
@akegan my thought for the review instructions is to test the scenario that led us to uncover the issue - testing that one can enter an address that isn't in SF but could be confused for being in SF, and checking that the address is not matched for any SF preference. i'll starting writing up review instructions.

**Pull request merged**